Makefile

```diff
@@ -102,3 +103,11 @@ gen_major:
 
 release_major: gen_major
 	git commit -m "bump to version $(shell cat version.md)"
+
+TESTS_PHP := $(shell find tests/Feature -name "*Test.php" -printf "%f\n")
+TEST_DONE := $(addprefix build/,$(TESTS_PHP:.php=.done))
```

Comment:
When I was debugging each tests, rather than executing the full test suite, I was running `make all_tests`.

TESTS_PHP contains a list of all the tests files = TESTS_PHP.
TEST_DONE generates a list of `build/testName.done` from TESTS_PHP

The recipe `all_tests` depends on all the TEST_DONE files and thus generates them.
those are generated with the recipe:
```makefile
build/%.done: tests/Feature/%.php
	vendor/bin/phpunit --filter $* && touch build/$*.done
```
which will run phpunit only for that specific test file and generate a done on success.

As a result, by doing `make all_tests` I will be executing all the tests, sequentially. This allows me to go through all the tests one by one and debug them each one at time rather than all in one go.
Modifying a test file also immediately triggers rebuilding the associated done file.